pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/MinionTaskUtils.java

```diff
@@ -67,7 +73,40 @@ static PinotFS getOutputPinotFS(Map<String, String> taskConfigs, URI fileURI)
     return PinotFSFactory.create(fileURIScheme);
   }
 
-  static PinotFS getLocalPinotFs() {
+  public static Map<String, String> getPushTaskConfig(String tableName, Map<String, String> taskConfigs,
+      ClusterInfoAccessor clusterInfoAccessor) {
+    try {
+      String pushMode = IngestionConfigUtils.getPushMode(taskConfigs);
+      Map<String, String> singleFileGenerationTaskConfig = new HashMap<>(taskConfigs);
+      if (pushMode == null || pushMode.contentEquals(BatchConfigProperties.SegmentPushType.TAR.toString())) {
+        singleFileGenerationTaskConfig.put(BatchConfigProperties.PUSH_MODE,
+            BatchConfigProperties.SegmentPushType.TAR.toString());
+      } else {
+        URI outputDirURI = URI.create(clusterInfoAccessor.getDataDir() + "/" + tableName);
+        String outputDirURIScheme = outputDirURI.getScheme();
+
+        if (!isLocalOutputDir(outputDirURIScheme)) {
+          singleFileGenerationTaskConfig.put(BatchConfigProperties.OUTPUT_SEGMENT_DIR_URI, outputDirURI.toString());
+          singleFileGenerationTaskConfig.put(BatchConfigProperties.PUSH_MODE, pushMode);
```

Comment:
nit: should we set push_mode to `metadata` when the provided one is `URI`? and we should log this change.